src/ape/managers/compilers.py

```diff
@@ -355,7 +355,7 @@ def flatten_contract(self, path: Path) -> Content:
             )
 
         compiler = self.registered_compilers[path.suffix]
-        return compiler.flatten_contract(path)
+        return compiler.flatten_contract(path, **kwargs)
```

Comment:
I'd also be happy to add `base_path: Optional[Path]` to `CompilerAPI` if that makes more sense.  But this way doesn't require changes to `ape_solidity`.